### DISCUSSION TITLE
PorousFlow Temperature Boundary Conditions with p-h formulation

### DISCUSSION THREAD

**Discussion by fparisio-zz** [idaholab/moose#16072] [Q&A Modules: General] [answered]
Dear all,
I am new to Moose and, more specifically, I am interested in PorousFlow. I would like to simulate a 2 phase flow of pure water with phase change and I am using a pressure-enthalpy formulation. My question is: how can I apply a temperature boundary condition? In other words, how to constrain enthalpy at a boundary such that, when the pressure changes, the temperature remains constant (or changes according to a specified value)?
Thank you for your help,

Francesco

**Comment by aeslaughter** [comment, marked as answer]
@WilkAndy Do you have any insight you could provide?

**Reply by WilkAndy** [reply]
Actually, i think @cpgr might be the expert here

**Reply by cpgr** [reply]
I'm not sure we can do this at the moment, but it wouldn't be hard to add and would definitely be useful.

**Reply by fparisio-zz** [reply]
Thank you @aeslaughter @WilkAndy @cpgr ! Should I open a new issue on this?

**Reply by aeslaughter** [reply]
@fparisio You can create an issue if you want, it is up to you.

**Comment by cpgr** [comment]
Hey @WilkAndy - didn't David create an enthalpy sink that should work here? I can't find it. Did it never make it into the repo?

**Reply by andrsd** [reply]
The PR got stale, because I got dragged into other shenanigans. I still plan to revive it and push it into the module.

**Comment by WilkAndy** [comment]
Yea, it might help here - i think @andrsd is still working on it.  In the case at hand, you have to change the enthalpy on the boundary as a function of porepressure.  As i recall, @andrsd was injecting a constant enthalpy source, which is a little different.

**Reply by andrsd** [reply]
My BC was taking pressure and temperature and computing `h` via fluid property look up. So yes, it is not exactly what @fparisio wants.

**Reply by fparisio-zz** [reply]
@andrsd that might actually do it if you can fix both p and T. I was thinking of something like PorousFlowFluidPropertyIC, where porepressure and temperature are used to compute enthalpy IC. One question: would that cause issues through phase change because of h jump?

**Reply by WilkAndy** [reply]
I can almost guarantee you'll get convergence problems around phase transitions - it always happens.  But that's just too bad if your system has phase transitions, and it should help that we've tried to make PorousFlow as robust as possible while still being fairly flexible.
